Fix appendModel issue when a frame have non null transformation

jorisv opened the pull request:
## Description

When a frame have inertia and a non null transformation, `appendModel` algorithm was doing some wrong computation to compute joint inertia of the new transform.

Close #2805 

## TODO
- [x] Fix `AppendJointOfModelAlgoTpl`
- [x] Fix  `appendUniverseToModel`
- [x] Check if there is an issue with eometry

## Checklist

- [x] I have run `pre-commit run --all-files` or `pixi run lint`
- [x] I have performed a self-review of my own code
- [x] I have commented my code where necessary
- [x] I have made corresponding changes to the doxygen documentation
- [x] I have added tests that prove my fix or feature works
- [x] I have updated the [CHANGELOG](https://github.com/stack-of-tasks/pinocchio/blob/devel/CHANGELOG.md) or added the "no changelog" label if it's a CI-related issue
- [x] I have updated the [README credits section](https://github.com/stack-of-tasks/pinocchio?tab=readme-ov-file#credits)

jorisv committed:
test: Add non regression test

jorisv committed:
core: Fix appendModel issue

jorisv committed:
changelog: Add entry

jorisv committed:
core: Add a comment

martisalcedo7 commented:
Shouldn't this be changed as well in the `appendUniverseToModel()` method?

jorisv commented:
@martisalcedo7 I'm currently looking at it but frame with inertia attached to universe is kind of weird. Do you see an use case of it ?

martisalcedo7 commented:
> @martisalcedo7 I'm currently looking at it but frame with inertia attached to universe is kind of weird. Do you see an use case of it ?

It is weird but it can be there. We have to make sure this inertia is not given to the body with addFrame(), therefore the argument append_inertia should be set to false.

jorisv commented:
Ok, after discussion it can model the inertia of the base of a manipulator.

jorisv committed:
test: Add non regression test for appendUniverseToModel

jorisv committed:
core: Fix appendUniverseToModel bug

martisalcedo7 commented:
Can we also apply the fix here for the geometric bodies?

They are expressed with respect to the frames, not with respect to the joints, so this transformation is wrong. It was also shown in the issue.

In the function `appendUniverseToModel()`

`go.placement = (pframe.placement * pfMAB) * go.placement;`

jorisv commented:
As said in the issue I'm waiting to have an example to reproduce it.

martisalcedo7 commented:
> As said in the issue I'm waiting to have an example to reproduce it.

I am working on that.

jorisv committed:
core: Avoid undefined behavior

`model.frames` vector is modified in the algorithm.
The `pframe` reference can then be invalidated.

martisalcedo7 commented:
It was a bug in our side. Please omit this geometry problem.

jorisv committed:
changelog: Improve entries

JTem commented:
i also confirm, ignore the geometry issue
